go1.18-openssl-fips: building toolchain2 fails with undefined reference to `dlsym'

I tried to build the new `go1.18-openssl-fips` branch on Fedora, with:
```console
GOROOT_BOOTSTRAP=/ GOROOT_FINAL=/usr/lib/golang GOHOSTOS=linux GOHOSTARCH=amd64 GOOS=linux GOARCH=amd64 ./make.bash -v
```
but got this error:
```console
Building Go toolchain2 using go_bootstrap and Go toolchain1.
...
# crypto/internal/boring
/usr/bin/ld: $WORK/b061/_x002.o: in function `_goboringcrypto_EVP_CIPHER_CTX_free':
/_/crypto/internal/boring/goopenssl.h:668: undefined reference to `dlsym'
/usr/bin/ld: $WORK/b061/_x002.o: in function `_goboringcrypto_EVP_CIPHER_CTX_new':
/_/crypto/internal/boring/goopenssl.h:287: undefined reference to `dlsym'
...
```
@derekparker any ideas? The same command succeeds with the `go1.17-openssl-fips` branch (3ece7d349521057450d587a94a495b2c476d0d88).

--- a/src/crypto/internal/boring/boring.go
+++ b/src/crypto/internal/boring/boring.go
@@ -8,8 +8,7 @@
 package boring
 
 /*
-// goboringcrypto_linux_amd64.syso references pthread functions.
-#cgo LDFLAGS: "-pthread"
+#cgo LDFLAGS: -ldl
 
 #include "goboringcrypto.h"
 */

